clinica/pipelines/engine.py

```diff
@@ -112,21 +112,18 @@ def __init__(
         )
         self._info = {}
 
-        if base_dir is None:
-            self.base_dir = mkdtemp()
-            self._base_dir_was_specified = False
-        else:
+        if base_dir:
             self.base_dir = base_dir
             self._base_dir_was_specified = True
-
-        if name:
-            self._name = name
         else:
-            self._name = self.__class__.__name__
+            self.base_dir = mkdtemp()
+            self._base_dir_was_specified = False
+
+        self._name = name or self.__class__.__name__
```

Comment:
I'm not familiar with this syntax. What's the purpose of this `or` between two instructions?

Reply:
I guess it's a shortcut to assign the variable whose value is not `None`. In cases both are not `None`, the first variable will be assigned.

Reply:
It's equivalent to a coalesce

Reply:
i.e. takes the first argument which is not None